app/src/main/java/org/schabi/newpipe/settings/domain/usecases/get_preference/GetPreference.kt

```diff
@@ -0,0 +1,7 @@
+package org.schabi.newpipe.settings.domain.usecases.get_preference
+
+import kotlinx.coroutines.flow.Flow
+
+fun interface GetPreference<T> {
+    operator fun invoke(key: Int, defaultValue: T): Flow<T>
+}
```

Comment:
Using [DataStore](https://developer.android.com/jetpack/androidx/releases/datastore) might be a better option instead of adding custom implementations for handling preferences.